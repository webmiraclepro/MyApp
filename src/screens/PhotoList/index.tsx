import React from 'react'
import { SafeAreaView } from 'react-navigation'
import { StatusBar, View } from 'react-native'
import { useNetInfo } from '@react-native-community/netinfo'
import LottieView from 'lottie-react-native'
import R from 'res'

import { MenuButton, PhotoCard } from '../../components'
import { CustomListContainer } from '../../containers'
import { usePhotoList } from '../../hooks'
import styles from './styles'

const PhotoListScreen = () => {
  const items = usePhotoList()
  const netInfo = useNetInfo()

  return (
    <>
      <StatusBar barStyle="dark-content" />
      <SafeAreaView style={styles.container} forceInset={{ bottom: 'always' }}>
        <CustomListContainer
          items={items}
          style={styles.content}
          renderItem={({ item }) => <PhotoCard {...item} />}
        />
      </SafeAreaView>
      <View style={styles.footer}>
        <LottieView
          autoSize
          autoPlay
          loop={false}
          style={styles.netInfo}
          resizeMode="contain"
          source={
            netInfo.isInternetReachable
              ? R.animations.online
              : R.animations.offline
          }
        />
      </View>
    </>
  )
}

PhotoListScreen.navigationOptions = (props: any) => {
  return {
    title: 'Photos',
    headerLeft: () => <MenuButton {...props} />,
  }
}

export default PhotoListScreen
