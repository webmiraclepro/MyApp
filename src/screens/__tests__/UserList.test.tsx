import React from 'react'
import { mount, shallow } from 'enzyme'
import PhotoList from '../PhotoList'

jest.mock('../../hooks', () => ({
  usePhotoList: jest.fn(() => []),
}))

describe('<PhotoList />', () => {
  const defaultProps = {
    navigation: jest.fn(),
  } as any

  it('should render', () => {
    const rendered = mount(<PhotoList {...defaultProps} />)
    expect(rendered).toBeTruthy()
  })

  it('renders correctly', () => {
    expect(shallow(<PhotoList {...defaultProps} />)).toMatchSnapshot()
  })
})
