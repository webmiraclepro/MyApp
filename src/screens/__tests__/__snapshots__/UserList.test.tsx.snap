// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserList /> renders correctly 1`] = `
<Fragment>
  <StatusBar
    animated={false}
    backgroundColor="black"
    barStyle="light-content"
    showHideTransition="fade"
  />
  <withOrientation
    forceInset={
      Object {
        "top": "always",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "backgroundColor": "#231E32",
        },
      ]
    }
  >
    <CustomListContainer
      items={
        Array [
          Object {
            "address": "street 26",
            "email": "jdnichollsc@hotmail.com",
            "first": "Juan",
            "last": "Nicholls",
          },
        ]
      }
      renderItem={[Function]}
      style={
        Object {
          "backgroundColor": "#231E32",
        }
      }
    />
  </withOrientation>
</Fragment>
`;
